includes/fbproduct.php

```diff
@@ -376,13 +376,13 @@ public function add_sale_price( $product_data, $for_items_batch = false ) {
 			intval( round( $this->get_price_plus_tax( $sale_price ) * 100 ) );
 
 			$sale_start =
-			( $date     = get_post_meta( $this->id, '_sale_price_dates_from', true ) )
-			? date_i18n( 'Y-m-d', $date ) . self::MIN_TIME
+			( $date     = $this->woo_product->get_date_on_sale_from() )
+			? date_i18n( 'Y-m-d', $date->getOffsetTimestamp() ) . self::MIN_TIME
```

Comment:
IMHO this change resets the local timezone and Facebook sale may start sooner or later than the original one depending on the timezone the store has. 

#### Example

I have my store in UTC+10:00 and I set Sale start `2022-05-15 00:00 UTC+10:00`. This is saved into database as `1652536800` timestamp, which is `2022-05-14 14:00:00 UTC+00:00`.

When posting to Facebook I get `1652536800` from the database and want to take a date `Y-m-d` out of it. And I get `2022-05-14`. Then I concatenate `T00:00+00:00` to it and send to Facebook. What I just did - I lost `14:00` hours. Jason had his Facebook Store set to `-08:00`, Facebook got our date `2022-05-14T00:00+00:00` and applied `-08:00` to it which resulted into `2022-05-13 16:00`. Which is kind of two days before we wanted it, right? 


#### Problem
The problem is not a timestamp we feed into `date_i18n`, but the concatenation ` . self::MIN/MAX_TIME`.

#### Solution
I would do it 

```suggestion
			? date_i18n( DATE_ATOM, $date->getOffsetTimestamp() )
```

or 

```suggestion
			? date_i18n( 'Y-m-d\TH:i:s+00:00', $date->getTimestamp() )
```

Reply:
You're spot on @message-dimke. Thanks for taking the time to look at this. 

PS: I learned about the DATE_ATOM constant.

Thanks,

R